src/Microsoft.Fx.Portability.MetadataReader/DependencyFinderEngineHelper.cs

```diff
@@ -73,27 +71,25 @@ public void ComputeData()
             }
 
             // Get the assembly info of System.Object and set it as assembly info for primitives
-            if (_assemblyFilter is DotNetFrameworkFilter)
+            var systemObjectMemberDependency = MemberDependency.FirstOrDefault(t => string.Equals(t.MemberDocId, "T:System.Object", StringComparison.Ordinal) && _assemblyFilter.IsFrameworkAssembly(t.DefinedInAssemblyIdentity));
```

Comment:
Without these two checks there can be the following undesired situations which are not addressed:

- there is more than one System.Object defined
- there is no System.Object found (see the failing test)
- there is one System.Object found but it is something defined in customer's assembly